### PR TITLE
Fix has many

### PR DESCRIPTION
**Pull request by dougmrqs**
Rebased and reviewed tests. Substitutes #68.

### CODE REVIEW

File: tests/query_interface_test.js [hunk 1]
```diff
@@ -440,7 +440,8 @@ describe('QueryInterface', () => {
     });
   });
 
-  describe('constraints', () => {
+  // Reason: gives this error on the before each: column "uid" does not exist. Works fine on CI
```
Comment (RichardJCai) [inline tests/query_interface_test.js:443]: Do you know why does this passes on CI but not locally?

File: source/index.js [hunk 2]
```diff
@@ -84,7 +84,7 @@ ConnectionManager.prototype.__loadDialectModule = ConnectionManager.prototype._l
 ConnectionManager.prototype._loadDialectModule = function (...args) {
   const pg = this.__loadDialectModule(...args);
   pg.types.setTypeParser(20, function (val) {
-    if (val > Number.MAX_SAFE_INTEGER) return BigInt(val);
+    if (val > Number.MAX_SAFE_INTEGER) return String(val);
```
Comment (RichardJCai) [inline source/index.js:87]: I just found out about this issue, sorry for catching this late:
https://github.com/sequelize/sequelize/issues/10468

It seems like from a ux perspective, sometimes users do expect BigInt and not string. I think we should aim to keep this as BigInt.

File: tests/has_many_test.js [hunk 3]
```diff
@@ -0,0 +1,1633 @@
...
+
+        expect(users[1].tasks[1].title).to.equal('b');
+        expect(users[1].tasks[1].subtasks.length).to.equal(2);
+        expect(users[1].tasks[1].subtasks[0].title).to.equal('b');
+        expect(users[1].tasks[1].subtasks[1].title).to.equal('a');
+      });
+
+      // Query result are non-deterministic
+      // Reason: https://github.com/sequelize/sequelize/issues/13088
+      it.skip('should fetch associations for multiple instances with limit and order and a belongsTo relation', async function() {
```
Comment (RichardJCai) [inline tests/has_many_test.js:282]: Could we unskip this by sorting the results array afterwards?

File: source/index.js (hunk 2 above)
Comment (dougmrqs) [reply source/index.js]: According to our last meeting, we've decided to trust in Sequelize's core. We'll keep treating those as Strings for now.

File: tests/has_many_test.js (hunk 3 above)
Comment (dougmrqs) [reply tests/has_many_test.js:282]: Sorting it afterwards just for the test to pass is not a good solution, in my opinion. I'll investigate further to see where exactly it is failing.

File: tests/query_interface_test.js (hunk 1 above)
Comment (dougmrqs) [reply tests/query_interface_test.js]: Removed unrelated skip.

**Review by RichardJCai — CHANGES_REQUESTED** [review]
Do you know why a lot of tests from https://github.com/sequelize/sequelize/blob/main/test/integration/associations/has-many.test.js don't seem to be here?

**Comment by dougmrqs** [comment]
> Do you know why a lot of tests from https://github.com/sequelize/sequelize/blob/main/test/integration/associations/has-many.test.js don't seem to be here?

Because they pass. I thought we would maintain only the broken ones to not have to keep a test suit of our own. We will trust Sequelize test suite. I've removed them in a commit, and I've been working this way, removing non-breaking tests. Does it make sense?

**Review by RichardJCai — APPROVED** [review]

**Pull request merged**